lib/modules/manager/crossplane/extract.ts

```diff
@@ -0,0 +1,53 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import type { XPKG } from './schema';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  extractConfig?: ExtractConfig,
+): PackageFileContent | null {
+  let definitions: XPKG[];
+  try {
+    definitions = loadAll(content) as XPKG[];
```

Comment:
Your are actually not using zod validation here see this example https://github.com/renovatebot/renovate/blob/d4db499554c57a944ae2022d67f5ff819c81b04f/lib/modules/manager/pep621/utils.ts#L117

Reply:
I gave it a shot, let me know if I got it correct!